{"CVE Code": "CVE-2022-41908", "Description": "TensorFlow is an open source platform for machine learning. An input `token` that is not a UTF-8 bytestring will trigger a `CHECK` fail in `tf.raw_ops.PyFunc`. We have patched the issue in GitHub commit 9f03a9d3bafe902c1e6beb105b2f24172f238645. The fix will be included in TensorFlow 2.11. We will also cherrypick this commit on TensorFlow 2.10.1, 2.9.3, and TensorFlow 2.8.4, as these are also affected and still in supported range.", "NVD Link": "https://nvd.nist.gov/vuln/detail/CVE-2022-41908", "Vulnerability Categories": "Input validation", "Affected Products": "\nGoogle\u00bbTensorflowVersions  from including (>=) 2.9.0andbefore (<) 2.9.3cpe:2.3:a:google:tensorflow:*:*:*:*:*:*:*:*Matching versions\nGoogle\u00bbTensorflowVersions  before (<) 2.8.4cpe:2.3:a:google:tensorflow:*:*:*:*:*:*:*:*Matching versions\nGoogle\u00bbTensorflow\u00bbVersion:2.10.0cpe:2.3:a:google:tensorflow:2.10.0:*:*:*:*:*:*:*Matching versions", "EPSS Score": "0.09% Probability of exploitation activity in the next 30 days\n~ 39 % Percentile, the proportion of vulnerabilities that are scored at or less", "EPSS History": "\n #  Date  Old EPSS Score  New EPSS Score  Delta (New - Old) \n 1  2024-12-17  0.10%  0.09%  -0.01 \n 2  2024-07-09  0.09%  0.10%  +0.02 \n 3  2024-06-25  0.07%  0.09%  +0.02 \n 4  2024-06-10  0.07%  0.07%  +0.00 \n 5  2024-02-23  0.07%  0.07%  0.00 \n 6  2023-12-21  0.06%  0.07%  +0.01 \n 7  2023-11-25  0.05%  0.06%  +0.01 \n 8  2023-10-19  0.05%  0.05%  0.00 \n 9  2023-03-07  0.95%  0.05%  -0.90 ", "CVSS Scores": "7.5", "CWE": [{"CWE_ID": "CWE-20 Improper Input Validation", "CWE_link": "/cwe-details/20/Improper-Input-Validation.html"}], "Reference": [{"ref_link": "https://github.com/tensorflow/tensorflow/security/advisories/GHSA-mv77-9g28-cwg3", "ref_desc": "`CHECK` fail via inputs in `PyFunc` \u00b7 Advisory \u00b7 tensorflow/tensorflow \u00b7 GitHubExploit;Patch;Third Party Advisory", "ref_cve_link": "N/A"}, {"ref_link": "https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/lib/core/py_func.cc", "ref_desc": "tensorflow/py_func.cc at master \u00b7 tensorflow/tensorflow \u00b7 GitHubThird Party AdvisoryCVEs referencing this url", "ref_cve_link": "https://www.cvedetails.com/reference-url-info/L3V6o5qYi9hf-5aB6nMBB2EJqiU.html"}, {"ref_link": "https://github.com/tensorflow/tensorflow/commit/9f03a9d3bafe902c1e6beb105b2f24172f238645", "ref_desc": "Replace CHECK with returning an InternalError on failing to create py\u2026 \u00b7 tensorflow/tensorflow@9f03a9d \u00b7 GitHubPatch;Third Party Advisory", "ref_cve_link": "N/A"}]}